src/util.h

```diff
@@ -677,6 +677,27 @@ enum ESkillLevel
     SL_PROFESSIONAL = 3
 };
 
+inline QString SkillLevelToString ( const int iSkill )
```

Comment:
Why is  this bound to `"CServerDlg"`?  Why not just re-use the existing look-up method?

Reply:
As a reminder - **_NO_** functionality - and _this_ counts as functionality - should exist "in the UI".  So explicitly making a second copy of something that should not exist really is bad...

Reply:
Ideally `server.cpp` would have a method to do the lookup that delegated to a shared utility method also used by `client.cpp`... although, in both cases, likely delegated via `channel.cpp`.

Reply:
> Why is this bound to `"CServerDlg"`? Why not just re-use the existing look-up method?

Perhaps I missed where this exists today... I'm happy to change it.

Reply:
> As a reminder - **_NO_** functionality - and _this_ counts as functionality - should exist "in the UI". So explicitly making a second copy of something that should not exist really is bad...

What is in the UI that you object to?   The change to CServerDlg is required because a method prototype was changed in CServer that is called **from** CServerDlg.   

And, again, if I missed an existing method for converting skill number to skill literal I will happily refactor -- but I didn't find one (perhaps I missed it?)

Reply:
UI is any user interface component.  That means the graphical user interface, the RPC interface and MIDI controls for the Jamulus core code.

Reply:
> And, again, if I missed an existing method for converting skill number to skill literal I will happily refactor -- but I didn't find one (perhaps I missed it?)

Like I say, there's a _lot_ of code in the UI that shouldn't be there.  Currently the mapping is done _literally_ in the UI: src/clientsettingsdlg.cpp lines 611 to 629.

I'm happy adding the method here, as you've done.  The first argument in the `QCoreApplication::translate` call (`"CServerDlg"` or whatever) doesn't seem used - I can't see anything in the *.ts files referencing it.  The *.ts files _do_ reference the containing file (`util.h` here).  I'd rather it just said "util", and we moved away from trying to guess where utility methods got called from...

Reply:
Sorry, but I do not know how to respond to this.  I don't see where there is "functionality" in the RPC code that is any different from the first release of RPC.  It merely calls the relevant methods elsewhere (just as the first version of RPC did).   And if you're only referring to skill code translation, even that is in util.* where other core code is -- not in the RPC code.   So I'm sorry @pljones, I'm not following your comments enough to know how to effect a change.

Reply:
`src/clientsettingsdlg.cpp` isn't the RPC code.  You're duplicating code here that's literally in the graphical user interface.  It shouldn't be.  That's what I'm complaining about.  The trigger for my complaint was the mention of the Service Graphical User Interface.  Which is irrelevant to what you're doing and hence _wrong_ and needs changing.

I've said that this is the _right_ place to do it.  Ideally the code in `src/clientsettingsdlg.cpp` would get fixed to call this method, too, and the translation would only happen in one place, consistently.

Reply:
Heh, I was confused about the subject of complaint as well as the primary indicator for the comment thread is... well, the new code, which was *not* the subject. ;)

Reply:
> The trigger for my complaint was the mention of the Server Graphical User Interface. Which is irrelevant to what you're doing and hence _wrong_ and needs changing.

we covered this here: https://github.com/jamulussoftware/jamulus/issues/2488#issuecomment-1065865193

The challenge with GetConCliParam is that it entirely uses pass-by-reference parameters.  I’m not sure it is reasonable to expect every caller to this method to have to use every reference returned.   Before this PR, the only place this method was called was from the server dialog.   I thought it was more efficient to modify and re-use this method instead of creating an entirely new one for RPC.   In fact, the server dialog might benefit now from the additional channel info that was never available to it before; I’ll look into this.